### PR TITLE
feat(vacuum): add brigade-vacuum

### PR DESCRIPTION
**Pull request by technosophos**
This cleans old builds.

A new CronJob is added that runs the brigade-vacuum every hour. By default, the vacuum cleans up any builds older than 30 days. It removes all of the pods and secrets associated with expired builds.

### CODE REVIEW

File: brigade-vacuum/cmd/brigade-vacuum/vacuum/vacuum_test.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	client := setupFakeClient()
+
+	secrets, err := client.CoreV1().Secrets(v1.NamespaceDefault).List(meta.ListOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(secrets.Items) != 3 {
+		t.Fatalf("expected 3 secrets, got %d", len(secrets.Items))
+	}
+	pods, err := client.CoreV1().Pods(v1.NamespaceDefault).List(meta.ListOptions{})
```
Comment (adamreese) [inline brigade-vacuum/cmd/brigade-vacuum/vacuum/vacuum_test.go]: err needs a home

File: brigade-vacuum/cmd/brigade-vacuum/vacuum/vacuum_test.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	if len(pods.Items) != 3 {
+		t.Fatalf("expected 3 pods, got %d", len(pods.Items))
+	}
+
+	num, err := New(time.Now(), 0, client, v1.NamespaceDefault).Run()
+	if err != nil {
+		t.Errorf("I blame fakeclient: %s", err)
```
Comment (adamreese) [inline brigade-vacuum/cmd/brigade-vacuum/vacuum/vacuum_test.go]: wasn't me

File: brigade-vacuum/cmd/brigade-vacuum/vacuum/vacuum_test.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+	}
+	if secrets.Items[0].Name != "scrooge" {
+		t.Errorf("expected scrooge to be the last secret, got %q", secrets.Items[0].Name)
+	}
+}
+
+// setupFakeClient creates a fake Kubernetes client with some "old" data.
+func setupFakeClient() kubernetes.Interface {
+	client := fake.NewSimpleClientset()
+	//client.PrependReactor("list", "secrets", secretsLister)
```
Comment (adamreese) [inline brigade-vacuum/cmd/brigade-vacuum/vacuum/vacuum_test.go]: delete me

File: brigade-vacuum/cmd/brigade-vacuum/vacuum/vacuum.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+	if v.max == 0 {
+		return deleted, nil
+	}
+
+	// We need to re-load the secrets list and see if we are still over the max.
+	secrets, err := v.client.CoreV1().Secrets(v.namespace).List(opts)
+	if err != nil {
+		return deleted, err
+	}
+	l := len(secrets.Items) //secrets.Size()
```
Comment (adamreese) [inline brigade-vacuum/cmd/brigade-vacuum/vacuum/vacuum.go]: delete comment

**Comment by technosophos** [comment]
Okay, I think I fixed all the issues, including the weird formatting.

**Pull request merged**